### DISCUSSION TITLE
Safety feature for shutters an blinds

### DISCUSSION THREAD

**Discussion by Ansgar66** [arendst/Tasmota#12381] [Ideas]
Dear Community,
I have an Idea that is triggered via a rollerblinds controller "Shutterbox" from blebox.eu, that might be based on an ESP as well.
This is able to learn the normal behaviour of a roller/blind by reading its power consumption over a complete down and up cycle. By interpretation of this curve it is able to set the most parameters like shutteropenduration/shuttersetopen etc..) correctly. 

But even further knowing the course of power during a standard course offers a great safety opportunity:
I fear to automatically run the shutter, especially closing, if I cannot watch the movement. If e.g. my wife left the balcony chair too close to the window, the shutter will rest on it when it is lowered. If it then does not switch off, the entire rest of the roller unwinds in the roller shutter box and a "roller shutter spaghetti" is created.

Switching off could be automized when the tasmota detects an unusual increase or decrease of power in comparison to the "normal" power for this position and direction of travel.

Would this be a feature to be implemented in Tasmota?

Best regards,
Ansgar

**Comment by TrDA-hab** [comment]
Hi!
1. We do not know anything about this device and its inner world.
2. This device has not been tested by anyone.
3. One of the developers has to buy a home for this device to carry out the tests.
4. Google doesn't know anything about blebox and ESP.
5. You must wait while your device is purchased and tested. And perhaps (!) your device will be added to the repository:
https://templates.blakadder.com/

Best regards,
TrDA

**Reply by sfromis** [reply]
Alternatively, you could dive into how the device works, and do the work to get it supported, if feasible.

**Comment by Ansgar66** [comment]
So I understand tue dual R3 being quote new and not completely 'tasmotized'.
So I can be happy to have it tasmotized somehow with the general firmware. 
I believe my idea is useful to be implemented for any shutter control device and will try my best to implement this feature myself.
But putting my first stets into tasmotas and having restricted programming knowledge it might take a while....

**Reply by barbudor** [reply]
I think this is a wrong statement.
AFAIK Dual R3 is fully supported by Tasmota regarding the existing feature set.
You are describing a feature that is not existing (yet) and as such is not supported on any device.
Moreover your original post was about a new and unknown device, not the Dual R3 so this is confusing.

**Comment by Ansgar66** [comment]
Hi Barbudor,
I try to be more precisely to awoid further missunderstandings:

1.) the alternative device "Shutterbox" mentioned above only triggered my Idea to copy similar and extended features to Tasmota-Hardware.

2.) I only guessed that the shutterbox might work on ESP-Hardware. The idea wasn't to alter this type of shutter controllers. The device is completely sealed; no possibility to flash or alter...

3.) I do not have deep programming knowledge an n'either knowledge on restrictions of the R3 hardware capabilities.
But I see the hardware offers all options to measure the electrical power course during an undisturbed shutter run. Comparing this course with the actual one could trigger an alarm or predescribed action in case values differ too much...

I hope this is less confusing now...

Best regards and thanks again!
Ansgar
